module/VuFind/src/VuFind/ILS/Driver/Folio.php

```diff
@@ -1692,13 +1807,18 @@ public function placeHold($holdDetails)
         // Account for an API spelling change introduced in mod-circulation v24:
         $fulfillmentKey = $this->getModuleMajorVersion('mod-circulation') >= 24
             ? 'fulfillmentPreference' : 'fulfilmentPreference';
+        $fulfillmentValue = $holdDetails['requestGroupId'] ?? 'Hold Shelf';
         $requestBody = $baseParams + [
             'requesterId' => $holdDetails['patron']['id'],
             'requestDate' => date('c'),
-            $fulfillmentKey => 'Hold Shelf',
+            $fulfillmentKey => $fulfillmentValue,
             'requestExpirationDate' => $requiredBy,
-            'pickupServicePointId' => $holdDetails['pickUpLocation'],
         ];
+        if ('Hold Shelf' == $fulfillmentValue) {
```

Comment:
Maybe rather than having this if/else down here, we could set a `$pickupLocationKey` above, similar to the way we do `$fulfillmentKey`. Of course there's nothing wrong with doing it this way if you prefer, I just thought that alternative might be slightly more concise (and would lend itself to using a `match` expression in future if a third case should crop up).

Reply:
I like the automatic exception throwing of `match`, was not familiar with that in PHP.  I've implemented that.